inputs/org.osbuild.containers-storage

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python3
+"""Inputs for container images from the host storage
+
+This is an input that pairs with the org.osbuild.containers-storage
+source. It makes the host container storage available to a stage
+and a container ID for the stage to use.
+
+Unlike all the other input types, this does not refer to static
+objects in the osbuild sources store but a dynamically mounted
+container store that depends on the availability of the
+resource (the specific container) on the host.
+"""
+
+import os
+import sys
+
+from osbuild import inputs
+from osbuild.util import containers
+from osbuild.util.mnt import MountGuard
+
+SCHEMA = r"""
+"definitions": {
+  "source-options": {
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "name": {
+        "type": "string",
+        "description": "The name to use for the image"
+      }
+    }
+  },
+  "source-object-ref": {
+    "type": "object",
+    "additionalProperties": false,
+    "minProperties": 1,
+    "patternProperties": {
+      ".*": {
+        "$ref": "#/definitions/source-options"
+      }
+    }
+  },
+  "source-array-ref": {
+    "type": "array",
+    "minItems": 1,
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["id"],
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "options": {
+          "$ref": "#/definitions/source-options"
+        }
+      }
+    }
+  },
+  "source-origin": {
+    "type": "string",
+    "description": "When the origin of the input is a source",
+    "enum": ["org.osbuild.source"]
+  }
+},
+"additionalProperties": false,
+"required": ["type", "origin", "references"],
+"properties": {
+  "type": {
+    "enum": ["org.osbuild.containers-storage"]
+  },
+  "origin": {
+    "description": "The org.osbuild.source origin case",
+    "$ref": "#/definitions/source-origin"
+  },
+  "references": {
+    "description": "Container image id",
+     "oneOf": [
+      {"$ref": "#/definitions/source-array-ref"},
+      {"$ref": "#/definitions/source-object-ref"}
+    ]
+  }
+}
+"""
+
+
+class ContainersInput(inputs.InputService):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.mg = MountGuard()
+        self.storage_conf = None
+
+    def map_storage_ref(self, _source, ref, target):
+        # we only want to read the host once and only for inputs that
```

Comment:
Is this comment/rational still true now that we have our own input type ? IIRC this was true when this code was part of the (bigger) `inputs/org.osbuild.containers` but now it's not anymore?

Reply:
Yeah this is an artifact from when we were doing everything in the same input.